Usiqee/Module/Artist/ArtistDetails/ArtistDetailsVC.swift

```diff
@@ -76,13 +89,84 @@ class ArtistDetailsVC: UIViewController {
     private func resetMenuSubViews() {
         discographyContainer.isHidden = true
     }
+    
+    private func handleFollowing() {
+        let isFollowing = ManagerAuth.shared.isFollowing(musicalEntity: musicalEntity)
+        if let artist = musicalEntity as? Artist {
+            if isFollowing {
+                followingButton.loadingIndicator(show: true)
+                unfollow(artist: artist)
+            } else {
+                followingButton.loadingIndicator(show: true, backgroundColor: Colors.purple)
+                follow(artist: artist)
+            }
+        } else if let band = musicalEntity as? Band {
+            if isFollowing {
+                followingButton.loadingIndicator(show: true)
+                unfollow(band: band)
+            } else {
+                followingButton.loadingIndicator(show: true, backgroundColor: Colors.purple)
+                follow(band: band)
+            }
+        }
+    }
+    
+    private func follow(artist: Artist) {
+        ServiceArtist.follow(artist: artist, completion: { [weak self] error in
+            self?.didFinishFollowing(error)
+        })
+    }
+    
+    private func unfollow(artist: Artist) {
+        guard let relatedArtist = ManagerAuth.shared.relatedArtist(by: artist.id!) else {
```

Comment:
starfoulah, we don't know if one day our dependencies will provide always an id, so can be check in the guard later

Reply:
mdrr le starfoulah ma tué xD I wanted to test if i works than i forget to changed it 🙃